apps/studio/prisma/schema.prisma

```diff
@@ -52,7 +52,7 @@ model Resource {
   siteId    Int
   site      Site      @relation(fields: [siteId], references: [id])
   parentId  BigInt?
-  parent    Resource? @relation("ParentRelation", fields: [parentId], references: [id])
+  parent    Resource? @relation("ParentRelation", fields: [parentId], references: [id], onDelete: Cascade)
```

Comment:
Are we doing hard delete or soft? Cascade will delete all right?

Reply:
We've never done soft delete and I'm not entirely sure on this because the extension that I found was using prisma client (which we're not, we're using kysely).

Because of this, I didn't delete the blob for auditing purposes and just kept it as is. 

Soft deletes, at least afaik, are done by the client library by inspecting a special column (`deletedAt` in classic) and hence, save devs the trouble of doing the querying etc themselves. 

We can do it here, but this would involve manual handling I think, and I think I'd rather keep this PR scoped to fixing this first rather than introducing another change!

Reply:
Am fine with this for now, but i do have concerns on hard deletes as we won't be able to surface full audit logs when required. We might want to file a ticket for exploring soft deletes

And yes i think manual handling would be required in this case but I think should be fine - just need to implement the deletedAt column and add a enum state for Resource to be in DELETED state

Reply:
filed here: https://linear.app/ogp/issue/ISOM-1541/implement-soft-deletes-for-studio